pyramda/iterable/flatten.py

```diff
@@ -1,37 +1,30 @@
 from collections import Iterable
 from past.builtins import basestring
 
+from pyramda.isinstance import isinstance
 from pyramda.function.curry import curry
 
 
 @curry
-def flatten(xs, depth):
+def flatten_until(is_leaf, xs):
```

Comment:
Should this work if the input `xs` isn't a list? Intuitively, it makes sense to me for `flatten_until(isinstance(basestring), 5)` to return `[5]`. What do you think?